File: backend/app/schemas/ranking.py
from pydantic import BaseModel

from app.schemas.prode_group import ProdeGroupRead
from app.schemas.user import UserRead


class RankingEntryRead(BaseModel):
    position: int

    user_id: int
    user: UserRead

    total_points: int
    predictions_count: int

    exact_scores_count: int
    winner_count: int
    goal_difference_count: int

    finished_predictions_count: int


class GroupRankingRead(BaseModel):
    group: ProdeGroupRead
    entries: list[RankingEntryRead]


class MyGroupRankingSummaryRead(BaseModel):
    group: ProdeGroupRead
    my_position: int | None
    my_points: int
    participants_count: int
    leader_name: str | None
    leader_points: int | None

File: frontend/src/types/ranking.ts
import type { ProdeGroup } from "./prodeGroup";
import type { User } from "./auth";

export interface RankingEntry {
  position: number;

  user_id: number;
  user: User;

  total_points: number;
  predictions_count: number;

  exact_scores_count: number;
  winner_count: number;
  goal_difference_count: number;

  finished_predictions_count: number;
}

export interface GroupRanking {
  group: ProdeGroup;
  entries: RankingEntry[];
}

export interface MyGroupRankingSummary {
  group: ProdeGroup;
  my_position: number | null;
  my_points: number;
  participants_count: number;
  leader_name: string | null;
  leader_points: number | null;
}